convert TZNG to ZSON in most ztests

nwt opened the pull request:
Exceptions:
* expr/ztests/cut-to-root.yaml (due to a ZSON bug)
* zio/detector/ztests/tzng-nonewline.yaml
* zio/detector/ztests/tzng.yaml
* zio/tzngio/ztests/*

mccanne reviewed (approved):
I hope this was automated :-)

nwt commented:
Mostly, with this.
```go
package main

import (
	"bytes"
	"fmt"
	"log"
	"os"
	"path/filepath"
	"regexp"
	"strings"

	"github.com/brimdata/zed/zbuf"
	"github.com/brimdata/zed/zio"
	"github.com/brimdata/zed/zio/tzngio"
	"github.com/brimdata/zed/zio/zsonio"
	"github.com/brimdata/zed/zng/resolver"
	"github.com/brimdata/zed/ztest"
	"gopkg.in/yaml.v3"
)

func main() {
	var paths []string
	if len(os.Args) > 1 {
		paths = os.Args[1:]
	} else {
		var err error
		paths, err = findZTests()
		if err != nil {
			log.Fatal(err)
		}
	}
	for _, path := range paths {
		zt, err := ztest.FromYAMLFile(path)
		if err != nil {
			log.Fatalf("%s: %s", path, err)
		}
		var modified bool
		if isValidTZNG(zt.Input) {
			if zt.Input, err = convertTZNGToZSON(zt.Input); err != nil {
				log.Fatalf("%s: %s", path, err)
			}
			modified = true
		}
		if isValidTZNG(zt.Output) {
			zt.OutputFlags = strings.ReplaceAll(zt.OutputFlags, "-f tzng", "")
			if zt.Output, err = convertTZNGToZSON(zt.Output); err != nil {
				log.Fatalf("%s: %s", path, err)
			}
			modified = true
		}
		for i, input := range zt.Inputs {
			if input.Data != nil && isValidTZNG(*input.Data) {
				s, err := convertTZNGToZSON(*input.Data)
				if err != nil {
					log.Fatalf("%s: %s", path, err)
				}
				zt.Inputs[i].Data = &s
				modified = true
			}
		}
		for i, output := range zt.Outputs {
			if output.Data != nil && isValidTZNG(*output.Data) {
				s, err := convertTZNGToZSON(*output.Data)
				if err != nil {
					log.Fatalf("%s: %s", path, err)
				}
				zt.Outputs[i].Data = &s
				modified = true
			}
		}
		if modified {
			b, err := formatYAML(zt)
			if err != nil {
				log.Fatalf("%s: %s", path, err)
			}
			if err := os.WriteFile(path, b, 0644); err != nil {
				log.Fatalf("%s: %s", path, err)
			}
		}
	}
}

func findZTests() ([]string, error) {
	pattern := fmt.Sprintf(`(^|\%c)ztests\%c.*\.yaml$`, filepath.Separator, filepath.Separator)
	re := regexp.MustCompile(pattern)
	var paths []string
	err := filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
		if !info.IsDir() && strings.HasSuffix(path, ".yaml") && re.MatchString(path) {
			paths = append(paths, path)
		}
		return err
	})
	return paths, err
}

func isValidTZNG(input string) bool {
	r := tzngio.NewReader(strings.NewReader(input), resolver.NewContext())
	var seen bool
	for {
		rec, err := r.Read()
		if err != nil {
			return false
		}
		if rec == nil {
			return seen
		}
		seen = true
	}
}

func convertTZNGToZSON(s string) (string, error) {
	zr := tzngio.NewReader(strings.NewReader(s), resolver.NewContext())
	var buf bytes.Buffer
	zw := zsonio.NewWriter(zio.NopCloser(&buf), zsonio.WriterOpts{})
	if err := zbuf.Copy(zw, zr); err != nil {
		return "", err
	}
	if err := zw.Close(); err != nil {
		return "", err
	}
	return buf.String(), nil
}

var formatRE = regexp.MustCompile("([^\n])\n(\\S)")

func formatYAML(v interface{}) ([]byte, error) {
	var buf bytes.Buffer
	e := yaml.NewEncoder(&buf)
	e.SetIndent(2)
	if err := e.Encode(v); err != nil {
		return nil, err
	}
	if err := e.Close(); err != nil {
		return nil, err
	}
	b := formatRE.ReplaceAll(buf.Bytes(), []byte("$1\n\n$2"))
	return b, nil
}
```

philrz reviewed (approved):
I didn't linger too long, but I swear I browsed through all the diffs! 😄
